mne_bids/write.py

```diff
@@ -2158,3 +2162,309 @@ def write_meg_crosstalk(fname, bids_path, verbose=None):
     logger.info(f'Writing crosstalk file to {out_path}')
     out_path.mkdir()
     shutil.copyfile(src=fname, dst=str(out_path))
+
+
+def _get_daysback(
+    *,
+    bids_path: BIDSPath,
+    rng: np.random.Generator
+) -> int:
+    """Try to find a suitable "daysback" for anonymization."""
+    raw = read_raw_bids(bids_path=bids_path)
+    daysback_min, daysback_max = get_anonymization_daysback([raw])
+    # We only estimated the range based on one file, so let's add a little
+    # slack here …
+    daysback_min += 356
+    daysback_max -= 365
+
+    # Pick one randomly
+    daysback = int(rng.choice(list(range(daysback_min, daysback_max + 1))))
+
+    return daysback
+
+
+@verbose
+def anonymize_dataset(bids_root_in, bids_root_out, daysback='auto',
+                      subject_mapping='auto', datatypes=None, rng_seed=None,
+                      verbose=None):
+    """Anonymize a BIDS dataset.
+
+    Parameters
+    ----------
+    bids_root_in : path-like
+        The root directory of the input BIDS dataset.
+    bids_root_out : path-like
+        The directory to place the anonymized dataset into.
+    daysback : int | 'auto'
+        Number of days by which to move back the recording date in time. If
+        ``'auto'``, tries to randomly pick a suitable number.
+    subject_mapping : dict | callable | 'auto' | None
+        How to anonymize subject IDs. If a dictionary, maps the original IDs
+        (keys) to the anonymized IDs (values). If a function, must be one that
+        accepts the original IDs as a list of strings and returns a dictionary
+        with original IDs as keys and anonymized IDs as values. If ``'auto'``,
+        automatically produces a mapping and prints it on the screen. If
+        ``None``, subject IDs are not changed.
+    datatypes : list of str | str | None
+        Which data type to anonymize. If can be ``meg``, ``eeg``, ``ieeg``, or
+        ``anat``. Multiple data types may be passed as a collection of strings.
+        If ``None``, try to anonymize the entire input dataset.
+    rng_seed : int | None
```

Comment:
don't be creative here :) it's called random_state everywhere in mne and use the check_random_state function to get an rng from it.

Reply:
Oh. I only found `seed` in the docdict and wasn't happy with it. Will have a look at what you proposed.